### ISSUE TITLE
Trying to examine function in stack trace, JsErrorDiagUnableToPerformAction

### ISSUE THREAD

**Issue by fatcerberus** [Microsoft/ChakraCore#3678]
I'm trying to examine function handles returned by JsDiagGetStackTrace(), so I can get the names of the functions to display in my command-line debugger.  However, doing this invariably results in JsDiagGetProperties() returning `JsErrorDiagUnableToPerformAction`.

**Comment by agarwal-sandeep** [comment]
@fatcerberus Based on code [JsDiagGetStackTrace](https://github.com/Microsoft/ChakraCore/blob/7bf217a285c2131ff1fbdb6989b73a356758d606/lib/Jsrt/JsrtDiag.cpp#L598) doesn't return JsErrorDiagUnableToPerformAction error.

Are you sure you are getting this error code from JsDiagGetStackTrace?

**Comment by fatcerberus** [comment]
No, the error is returned by JsDiagGetProperties() when I try to examine the function handle from the stack trace.

**Comment by fatcerberus** [comment]
Best I can tell, this is a bug, JsDiagGetStackTrace() returns:
```c
    ///     [{
    ///         "index" : 0,
    ///         "scriptId" : 2,
    ///         "line" : 3,
    ///         "column" : 0,
    ///         "sourceLength" : 9,
    ///         "sourceText" : "var x = 1",
    ///         "functionHandle" : 1
    ///     }]
```

However, passing the value of `functionHandle` to JsDiagGetProperties() immediately afterward results in JsErrorDiagUnableToPerformAction.  I'm trying to read the `.name` of the function to display in my debugger, but the error prevents me from doing so 🙁

**Comment by agarwal-sandeep** [comment]
@fatcerberus Yes it seems like a bug as JsrtDebuggerObjectFunction doesn't have a GetChildren implementation. Marking as bug and assigning to @digitalinfinity

**Comment by fatcerberus** [comment]
Is there any other way I can get the names of the functions in the stack trace?  I currently have to list all frames as anonymous functions in my debugger even though they're not actually anonymous :)

**Comment by agarwal-sandeep** [comment]
Can you call JsDiagGetObjectFromHandle(functionHandle) and use .name on it? That's what ch [debugger](https://github.com/Microsoft/ChakraCore/blob/9e0f8e466f81bf69c4a5f8cb547f3b4faf6a97a6/bin/ch/DbgController.js#L724) is doing.

**Comment by fatcerberus** [comment]
Oh, good idea, I didn't think of that.  I'll try it, thanks!

**Comment by fatcerberus** [comment]
> Can you call JsDiagGetObjectFromHandle(functionHandle) and use .name on it?

@agarwal-sandeep Thanks you, this worked.  Now SSJ's backtraces have function names again. 😄

**Comment by EdMaurer** [comment]
Clear assignment. Existence of work around lowers priority.

**Comment by agarwal-sandeep** [comment]
@EdMaurer The fix/enhancements is already in PR #3703

**Comment by agarwal-sandeep** [comment]
Closing as this is fixed and available in master.
